tests/unit/checkpoint/test_pipeline.py

```diff
@@ -0,0 +1,119 @@
+from deepspeed.runtime.checkpoint_engine.torch_checkpoint_engine import TorchCheckpointEngine
+from tests.unit.common import DistributedTest
+from tests.unit.simple_model import *
+
+from tests.unit.checkpoint.common import checkpoint_correctness_verification
+
+import pytest
+
+
+@pytest.fixture(params=[2])
+def stages(request):
+    return request.param
+
+
+class TestPipelineCheckpoint(DistributedTest):
+    world_size = 4
+
+    @pytest.mark.parametrize("zero_stage", [0, 1])
+    def test_checkpoint_pipe_engine(self, zero_stage, tmpdir, stages):
+        config_dict = {
+            "train_batch_size": 2,
+            "train_micro_batch_size_per_gpu": 1,
+            "steps_per_print": 1,
+            "optimizer": {
+                "type": "Adam",
+                "params": {
+                    "lr": 1e-5
+                }
+            },
+            "zero_optimization": {
+                "stage": zero_stage
+            },
+            "fp16": {
+                "enabled": zero_stage > 0
+            },
+            "scheduler": {
+                "type": "OneCycle",
+                "params": {
+                    "cycle_first_step_size": 1000,
+                    "cycle_first_stair_count": 500,
+                    "cycle_second_step_size": 1000,
+                    "cycle_second_stair_count": 500,
+                    "decay_step_size": 1000,
+                    "cycle_min_lr": 0.0001,
+                    "cycle_max_lr": 0.0010,
+                    "decay_lr_rate": 0.001,
+                    "cycle_min_mom": 0.85,
+                    "cycle_max_mom": 0.99,
+                    "decay_mom_rate": 0.0
+                }
+            }
+        }
+
+        def _test(save_folder, num_stages):
+            args = args_from_dict(tmpdir, config_dict)
+            models = [LinearStackPipe(num_stages=num_stages) for _ in range(2)]
+            checkpoint_correctness_verification(args=args,
+                                                models=models,
+                                                hidden_dim=models[0].hidden_dim,
+                                                tmpdir=save_folder,
+                                                fp16=config_dict['fp16']['enabled'],
+                                                load_optimizer_states=True,
+                                                load_lr_scheduler_states=True,
+                                                train_batch=True)
+
+        _test(tmpdir, num_stages=stages)
+
+    @pytest.mark.parametrize(
+        "base_topo,test_topo",
+        [
+            #(PipeTopo(num_pp=1,
+            #          num_dp=4),
+            # PipeTopo(num_pp=4,
+            #          num_dp=1)),
+            #(PipeTopo(num_pp=2,
+            #          num_dp=2),
+            # PipeTopo(num_pp=2,
+            #          num_dp=2)),
+            #(PipeTopo(num_pp=4,
+            #          num_dp=1),
+            # PipeTopo(num_pp=2,
+            #          num_dp=2)),
```

Comment:
If these are all supposed to be commented out, this parametrize is not needed

Reply:
Yes, it is currently all [commented out](https://github.com/microsoft/DeepSpeed/blob/f82846d7df6a129668bea447a0f953f6cf345efb/tests/unit/test_checkpointing.py#L712).

I am not sure the reason, but I seem to recall this being a flaky unit test.